services/identity/UsersService.ts

```diff
@@ -88,10 +83,18 @@ class UsersService {
     })
   }
 
-  canSendEmailOtp(email: string) {
+  async canSendEmailOtp(email: string) {
+    const whitelistEntries = await this.whitelist.findAll({
+      attributes: ["email"],
+      where: {
+        expiry: {
+          [Op.or]: [{ [Op.is]: null }, { [Op.gt]: new Date() }],
+        },
+      },
+    })
+    const whitelistDomains = whitelistEntries.map((entry) => entry.email)
     const hasMatchDomain =
-      this.whitelistDomains.filter((domain) => email.endsWith(domain)).length >
-      0
+      whitelistDomains.filter((domain) => email.endsWith(domain)).length > 0
```

Comment:
this is fine as-is and it might be over-eager from my end, but wdyt about having a `WhitelistService` that 

1. returns valid whitelists 
2. checks if a given email is valid

Reply:
As discussed offline, we currently don't foresee the whitelist being used anywhere other than login - as such, we'll leave it as is for now